webpack/assets/javascripts/react_app/components/common/Fill/FillReducer.js

```diff
@@ -0,0 +1,27 @@
+import Immutable from 'seamless-immutable';
+
+import { REGISTER_FILL, REMOVE_FILLED_COMPONENT } from './FillConstants';
+
+const initialState = Immutable({
+  RegisteredComponent: {},
```

Comment:
IMO `RegisteredComponent` is redundant, it could be:
```jsx
const initialState = Immutable({});

return state.setIn([payload.slotId, payload.fillId], true);
```

Reply:
?